### PR TITLE
Concourse contract tests

### PR DESCRIPTION
**Pull request by alice-carr**
This PR breaks out contract tests to a separate task in concourse and runs them every hour

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,5 @@ output/*
 htmlcov
 .coverage
 pipeline_vars.yml
+
```
Comment (0atman) [inline .gitignore]: empty line?

File: build/concourse/pipeline.yml [hunk 2]
```diff
@@ -33,6 +35,7 @@ resources:
       paths:
         - service/github_audit/account/779799343306/apply.tfvars
         - service/github_audit/account/779799343306/backend.tfvars
+
```
Comment (0atman) [inline build/concourse/pipeline.yml]: are these empty lines intentional?

File: config.py [hunk 3]
```diff
@@ -13,13 +13,13 @@ def set_region(region):
     REGION = region
 
 
-def load(env=None):
+def load():
     """
     Load settings
```
Comment (0atman) [inline config.py]: now I read this docstring properly, I hate it

File: config.py [hunk 4]
```diff
@@ -34,6 +34,7 @@ def get_ssm_client():
             region_name=region,
         )
     else:
+        print("aws_secret_key not in env, setting to none")
```
Comment (0atman) [inline config.py]: love

File: contract_tests/test_contract.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    """
+
+    for query in ["all", "prs", "refs"]:
+        page = Dict(pgraph.query(query, org="alphagov", nth=1, after=None))
+        assert page.organization.repositories.nodes
+
+
+def test_github_dependabot() -> None:
+    """
+    Tests that there are more than 10 active, inactive, by_status and custom, repos.
```
Comment (0atman) [inline contract_tests/test_contract.py]: we should update this docstring

File: test.sh [hunk 6]
```diff
@@ -10,6 +12,13 @@ for i in ${vars}; do
     fi
 done
 
+if [ ${directory}=="contract_tests" ]
+then
+    coverage_failure=""
+else
+    coverage_failure="--cov-fail-under 20"
+fi
+
```
Comment (0atman) [inline test.sh]: very spicy :ok_hand:

File: build/concourse/pipeline.yml (hunk 2 above)
Comment (alice-carr) [reply build/concourse/pipeline.yml]: they are indeed - we've been separating out tasks/job with a space in between each so I was just making this consistent with what we've already got

File: .gitignore [hunk 7]
```diff
@@ -23,3 +23,5 @@ output/*
...
+build/concourse/pipeline_var.yml
```
Comment (akinnane) [inline .gitignore]: Should this file end with a blank new line?

File: config.py [hunk 8]
```diff
@@ -13,13 +13,13 @@ def set_region(region):
...
 
 
-def load(env=None):
+def load():
     """
-    Load settings
+    Loads the correct settings file based on FLASK_ENV.
     """
-    env = env or os.environ.get("FLASK_ENV", "development")
+    flask_env = os.environ["FLASK_ENV"]
```
Comment (akinnane) [inline config.py]: What happens if `FLASK_ENV` isn't set?

File: dependabot_api.py [hunk 9]
```diff
@@ -22,7 +22,7 @@ def get_parsed(path):
     return body.data
 
 
-def get_repos_by_status(org):
+def get_active_repos(org):
```
Comment (akinnane) [inline dependabot_api.py]: Good move!

File: .gitignore (hunk 7 above)
Comment (alice-carr) [reply .gitignore]: I can add one in

Comment (0atman) [reply .gitignore]: Aye. All files, by definition, should end with a newline, as unix format dictates/

File: config.py (hunk 8 above)
Comment (0atman) [reply config.py]: A very readable exception happens, as it should.

**Pull request merged**